src/Microsoft.AspNet.Mvc.Core/Controllers/DefaultControllerActionArgumentBinder.cs

```diff
@@ -64,6 +64,28 @@ public class DefaultControllerActionArgumentBinder : IControllerActionArgumentBi
                         nameof(actionContext));
             }
 
+            if (actionDescriptor.BoundProperties.Count == 0 &&
+                actionDescriptor.Parameters.Count == 0)
+            {
+                return Task.FromResult<IDictionary<string, object>>(
+                    new Dictionary<string, object>(StringComparer.Ordinal));
```

Comment:
Why allocate the `Dictionary` or the `Task` in this mainline case? Suggest sticking the `Task` in a `static` field.